apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java

```diff
@@ -178,6 +190,9 @@ public void doEnd(long epochMicros) {
         if (type == null) {
             type = "custom";
         }
+        if (parent != null) {
+            parent.onChildEnd(this, epochMicros);
+        }
         this.tracer.endSpan(this);
```

Comment:
But a span can end before the reference count is 0.

Reply:
Sorry, I meant the other things that happen in end except from the timestamping, ie the parent reference update and the queueing for serialization and sending.
In other words, I suggested a seamless buffering of referenced spans without actually managing a buffer. This opens all kind of options for us for retrospective decisions. The problem with that is missed reference accounting (this is real memory leak) and very long waiting tasks like messaging or timers etc. This can be solved with timeouts, but surely complicates things.

Reply:
We can always add that later if needed. Let's defer this for now.